pandas/tests/frame/test_indexing.py

```diff
@@ -1940,6 +1940,16 @@ def test_setitem_with_unaligned_sparse_value(self):
         exp = pd.Series([1, 0, 0], name='new_column')
         assert_series_equal(df['new_column'], exp)
 
+    def test_setitem_with_unaligned_tz_aware_datetime_column(self):
```

Comment:
can you also test this will ``.loc`` and a mask

e.g.

``df.loc[[0,1,2],'dates'] = columns[[1,0,2]]`` (should be the same result)